connect-inject/container_init.go

```diff
@@ -206,7 +206,7 @@ services {
 
   proxy {
     destination_service_name = "{{ .ServiceName }}"
-    destination_service_id = "{{ .ServiceName }}"
+    destination_service_id = "${POD_NAME}-{{ .ServiceName }}"
```

Comment:
I think we should make `ServiceID` part of the struct that's used in this template. That's why this bug happened, because the ServiceID was being constructed inside the template and we missed copy pasting it.

Reply:
It makes sense

Reply:
I applied change to proxy service id as well